crates/library/std/src/evm.fe

```diff
@@ -0,0 +1,293 @@
+    return __number()
+
+pub fn difficulty() -> u256:
+  unsafe:
+    return __difficulty()
+
+pub fn gas_limit() -> u256:
+  unsafe:
+    return __gaslimit()
+
+pub fn gas() -> u256:
+  unsafe:
+    return __gas()
+
+pub fn self_address() -> address:
+  unsafe:
+    return address(__address())
+
+pub fn balance_of(addr: address) -> u256:
+  unsafe:
+    return __balance(u256(addr))
+
+pub fn balance() -> u256:
+  unsafe:
+    return __selfbalance()
+
+pub fn caller() -> address:
+  unsafe:
+    return address(__caller())
+
+pub fn call_value() -> u256:
+  unsafe:
+    return __callvalue()
+
+
+# Overflowing math ops. Should these be unsafe or named
+# `overflowing_add`, etc?
+pub fn add(x: u256, y: u256) -> u256:
+  unsafe:
+    return __add(x, y)
+
+pub fn sub(x: u256, y: u256) -> u256:
+  unsafe:
+    return __sub(x, y)
+
+pub fn mul(x: u256, y: u256) -> u256:
+  unsafe:
+    return __mul(x, y)
+
+pub fn div(x: u256, y: u256) -> u256:
+  unsafe:
+    return __div(x, y)
+
+pub fn sdiv(x: u256, y: u256) -> u256:
+  unsafe:
+    return __sdiv(x, y)
+
+pub fn mod(x: u256, y: u256) -> u256:
+  unsafe:
+    return __mod(x, y)
+
+pub fn smod(x: u256, y: u256) -> u256:
+  unsafe:
+    return __smod(x, y)
+
+pub fn exp(x: u256, y: u256) -> u256:
+  unsafe:
+    return __exp(x, y)
+
+pub fn addmod(x: u256, y: u256, m: u256) -> u256:
+  unsafe:
+    return __addmod(x, y, m)
+
+pub fn mulmod(x: u256, y: u256, m: u256) -> u256:
+  unsafe:
+    return __mulmod(x, y, m)
+
+pub fn sign_extend(i: u256, x: u256) -> u256:
+  unsafe:
+    return __signextend(i, x)
+
+
+
+# Comparison ops
+# TODO: return bool (see issue #653)
+pub fn lt(x: u256, y: u256) -> u256:
+  unsafe:
+    return __lt(x, y)
+
+pub fn gt(x: u256, y: u256) -> u256:
+  unsafe:
+    return __gt(x, y)
+
+pub fn slt(x: u256, y: u256) -> u256:
+  unsafe:
+    return __slt(x, y)
+
+pub fn sgt(x: u256, y: u256) -> u256:
+  unsafe:
+    return __sgt(x, y)
+
+pub fn eq(x: u256, y: u256) -> u256:
+  unsafe:
+    return __eq(x, y)
+
+pub fn is_zero(x: u256) -> u256:
+  unsafe:
+    return __iszero(x)
+
+
+# Bitwise ops
+pub fn bitwise_and(x: u256, y: u256) -> u256:
+  unsafe:
+    return __and(x, y)
+
+pub fn bitwise_or(x: u256, y: u256) -> u256:
+  unsafe:
+    return __or(x, y)
+
+pub fn bitwise_not(x: u256) -> u256:
+  unsafe:
+    return __not(x)
+
+pub fn xor(x: u256, y: u256) -> u256:
+  unsafe:
+    return __xor(x, y)
+
+pub fn byte(offset: u256, x: u256) -> u256:
+  unsafe:
+    return __byte(offset, x)
+
+pub fn shl(x: u256, y: u256) -> u256:
+  unsafe:
+    return __shl(x, y)
+
+pub fn shr(x: u256, y: u256) -> u256:
+  unsafe:
+    return __shr(x, y)
+
+pub fn sar(x: u256, y: u256) -> u256:
+  unsafe:
+    return __sar(x, y)
+
+
+# Evm state access and control
+pub unsafe fn return_mem(location: u256, len: u256):
+    return __return(location, len)
+
+pub unsafe fn revert_mem(location: u256, len: u256):
+    return __revert(location, len)
+
+pub unsafe fn self_destruct(addr: address):
```

Comment:
The underscore between `self` and `destruct` stands out odd with me.

Reply:
Ok, I'll remove.